src/System.IO.FileSystem/tests/File/Delete.cs

```diff
@@ -82,6 +82,27 @@ public void ShouldThrowIOExceptionDeletingDirectory()
             Assert.Throws<UnauthorizedAccessException>(() => Delete(TestDirectory));
         }
 
+        [ConditionalFact(nameof(CanCreateSymbolicLinks))]
+        public void DeletingSymLinkDoesntDeleteTarget()
+        {
+            var path = GetTestFilePath();
+            var linkPath = GetTestFilePath();
+
+            File.Create(path).Dispose();
+            Assert.True(MountHelper.CreateSymbolicLink(linkPath, path, isDirectory: false));
+
+            // Both the symlink and the target exist
+            Assert.True(File.Exists(path), "path should exist");
+            Assert.True(File.Exists(linkPath), "linkPath should exist");
+
+            // Delete the symlink
+            File.Delete(linkPath);
+
+            // Target should still exist
+            Assert.True(File.Exists(path), "path should still exist");
+            Assert.False(File.Exists(linkPath), "linkPath should no longer exist");
+        }
```

Comment:
Do we want a directory version of this, too? (Maybe it exists and I just haven't seen it yet?)